iidm-api/src/main/java/eu/itesla_project/iidm/network/util/Networks.java

```diff
@@ -414,24 +486,53 @@ public static boolean checkFlows(TwoWindingsTransformer twt) {
+            return checkFlows(network, threshold, verbose, formatter);
         }
-        return ok;
+    }
+
+    private static boolean checkFlows(Network network, float threshold, boolean verbose, TableFormatter formatter) {
+        LOGGER.info("Checking flows of network {}", network.getId());
+        boolean linesOk = Iterators.size(network.getLines().iterator()) == StreamSupport.stream(network.getLines().spliterator(), false)
+                                                                                         .sorted((l1, l2) -> l1.getId().compareTo(l2.getId()))
+                                                                                         .filter(l -> {
+                                                                                             try {
+                                                                                                 return checkFlows(l, threshold, verbose, formatter);
+                                                                                             } catch (IOException e) {
+                                                                                                 throw new RuntimeException(e);
+                                                                                             }
+                                                                                         })
+                                                                                         .count(); 
+        boolean transformersOk =  Iterators.size(network.getTwoWindingsTransformers().iterator()) == StreamSupport.stream(network.getTwoWindingsTransformers().spliterator(), false)
+                                                                                                                  .sorted((twt1, twt2) -> twt1.getId().compareTo(twt2.getId()))
+                                                                                                                  .filter(twt -> {
+                                                                                                                      try {
+                                                                                                                          return checkFlows(twt, threshold, verbose, formatter);
+                                                                                                                      } catch (IOException e) {
+                                                                                                                          throw new RuntimeException(e);
+                                                                                                                      }
+                                                                                                                  })
+                                                                                                                  .count();
+        return linesOk && transformersOk;
```

Comment:
I'm not sure this code is better than a classic for loop? Why don't you simply add a try/catch block in the for loop?